# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

base_startup_appspawn_standard: disable longProcName resetting

There's memory corruption happening in procedure (strncpy_s call) of resetting
the longProcName which in fact is `argv`. No traces of this actually being
needed for anything could be found. Question regarding this was asked here:
https://gitee.com/openharmony/startup_appspawn/pulls/308

TODO: revisit this patch once we have more information

Signed-off-by: Robert Drab <robert.drab@huawei.com>
Relates to https://gitlab.eclipse.org/eclipse/oniro-oh/meta-openharmony/-/issues/42
Upstream-Status: Pending

diff --git a/src/appspawn_server.cpp b/src/appspawn_server.cpp
index 2c8e7d1..18d9778 100644
--- a/src/appspawn_server.cpp
+++ b/src/appspawn_server.cpp
@@ -261,18 +261,21 @@ int32_t AppSpawnServer::SetProcessName(
         return (-errno);
     }
 
+#if 1
+    HiLog::Warn(LABEL, "logProcName resetting disabled!");
+#else
     // reset longProcName
     if (memset_s(longProcName, static_cast<size_t>(longProcNameLen), 0, static_cast<size_t>(longProcNameLen)) != EOK) {
         HiLog::Error(LABEL, "Failed to memset long process name");
         return -EINVAL;
     }
 
     // set long process name
     if (strncpy_s(longProcName, len, processName, len) != EOK) {
         HiLog::Error(LABEL, "strncpy_s long name error: %{public}d", strerror_r(errno, err_string, ERR_STRING_SZ));
         return -EINVAL;
     }
-
+#endif
     return ERR_OK;
 }
 
